### ISSUE TITLE
Option to disable everything except syntax highlighting, autocompletion, snippets and linters

### ISSUE THREAD

**Issue by dahu33** [Azure/vscode-kubernetes-tools#839]
I really don't want my VSCode to access my kube config / k8s clusters, however I would like to keep the syntax highlighting, autocompletion, snippets and linters that this extension provides. So far I couldn't find an option or an alternative extension that would do the job...

**Comment by lstocchi** [comment]
Hi @dahu33 , sorry for the late response. One thing i can think of is to setup a wrong path for your kube config so that vscode will point to a wrong file (settings.json -> then add `"vs-kubernetes.kubeconfig": "point-to-some-wrong-file"`). This way you could keep using all the snippets/autocompletion/linters without providing access to the correct config to vscode. Does it solve your issue?

**Comment by dahu33** [comment]
Yes and no. I strongly believe this should be a feature not a hack. Unfortunately there isn't many viable alternative extension for syntax highlighting only and I think a lot of people do not need more than that.

**Comment by dahu33** [comment]
> settings.json -> then add `"vs-kubernetes.kubeconfig": "point-to-some-wrong-file"`

the workaround doesn't even work, seems like it always fallback to the default kube config path. Also, looks like the plugin is trying to access the Kubernetes cluster defined in the current context every time I open a yaml file in vscode. I don't understand how people are confortable with that... I'm not but, I love the syntax highlighting and autocompletion that this plugin brings... Would be nice to have a solution in place.

**Comment by Lexicality** [comment]
I have to restart my WSL container after doing any k8s operations because this extension causes all sorts of weird behaviours when it attempts to connect to the cluster after my extremely short access token has expired.

I use this extension exclusively for syntax highlighting and automatic formatting and it would be great if I could configure it so that it doesn't issue _**ANY**_ kubectl commands without me opening the extension's tab and clicking a button.

**Comment by dfsdevops** [comment]
This issue is especially annoying when dealing with oidc-login setups because it will constantly hijack your browser to initiate oauth flow for any contexts in your config. Really need a way to just disable that feature altogether

**Comment by fhemberger** [comment]
> This issue is especially annoying when dealing with oidc-login setups because it will constantly hijack your browser to initiate oauth flow for any contexts in your config.
THIS.

The extension seems to be calling `kubectl view config -o json` under the hood to get the list of clusters. Added `"vscode-kubernetes.kubectl-path": "/usr/local/bin/vscode-kubectl-stub"` to the settings and created the stub:

```
echo echo \"{}\" > /usr/local/bin/vscode-kubectl-stub
chmod +x /usr/local/bin/vscode-kubectl-stub
```

Now the command just returns an empty JSON object, so no clusters are found by the extension.

**Comment by dynajoe** [comment]
+1 to how frustrating it is with oidc login and my browser constantly getting opened.
